vignettes/speedup_options.Rmd

```diff
@@ -0,0 +1,710 @@
+reported_cases_true <- posterior_sample[
+  variable == "reported_cases", .(date, confirm = value)
+]
+```
+
+Now, to the main part of this vignette: we will define and run the different model options and evaluate their runtimes and estimation and forecasting performance.
+
+## Model options
+
+Below we describe each model.
+
+
+Table: Model options
+
+|model               |description                                                                                                            |
+|:-------------------|:----------------------------------------------------------------------------------------------------------------------|
+|default_mcmc_rstan  |Default model; Gaussian process on new infection and with a non-stationary prior on $R_t$                              |
+|non_mechanistic     |Gaussian process on new infection and with no mechanistic process on $R_t$                                             |
+|no_rw_gp            |Gaussian process on new infection and with a stationary Gaussian process prior on $R_t$                                |
+|rw1_no_gp           |No Gaussian process on new infections, and $R_t$ modelled with 1-day random walk and a non-stationary Gaussian process |
+|rw1_gp              |Gaussian process on new infection, and $R_t$ modelled with 1-day random walk and a non-stationary Gaussian process     |
+|rw7_no_gp           |No Gaussian process on new infection, and $R_t$ modelled with 7-day random walk and a non-stationary process           |
+|rw7_gp              |Gaussian process on new infection, and $R_t$ modelled with 7-day random walk and a non-stationary Gaussian process     |
+|default_vb_rstan    |Default model; Variational algorithm for sampling instead of MCMC and using [`{rstan}`]                                |
+|default_vb_cmdstanr |Default model; Variational algorithm for sampling instead of MCMC and using [`{cmdstanr}`]                             |
+
+
+
+
+
+Table: Model components
+
+|model               |gaussian_process |rt_model_rw |rt_model_gp    |sampling              |package  |
+|:-------------------|:----------------|:-----------|:--------------|:---------------------|:--------|
+|default_mcmc_rstan  |on               |None        |non_stationary |MCMC                  |rstan    |
+|non_mechanistic     |on               |None        |None           |MCMC                  |rstan    |
+|no_rw_gp            |on               |None        |stationary     |MCMC                  |rstan    |
+|rw1_no_gp           |off              |1-day       |non_stationary |MCMC                  |rstan    |
+|rw1_gp              |on               |1-day       |non_stationary |MCMC                  |rstan    |
+|rw7_no_gp           |off              |7-days      |non_stationary |MCMC                  |rstan    |
+|rw7_gp              |on               |7-days      |non_stationary |MCMC                  |rstan    |
```

Comment:
is this suggesting having both a random walk on Rt and a GP? I think that might be very hard to fit?